udfs/tests/run.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#########################################################################
+# Call run.sh locally with an optional parameter specifying the test name
+# if you want to run only that test and no others
+#
+# e.g. bash udfs/tests/run.sh test_community_udf
+#
+#########################################################################
+python3 -m pip install -r udfs/tests/requirements.txt
```

Comment:
Installation of dependencies should happen outside of this script so the installation of dependencies doesn't happen on every test run.
locally I shouldn't have to run this pip step every time I want to run the tests just to functionally no-op.

I think you did this so you can just use the python slim image directly.

It would be ideal to use the Docker cloud builder to build an image (Dockerfile base image python slim + this pip install line) or pull from cache in gcr that has these deps installed and the cloud build should just use this as your base image. We could make this publicly available gcr image for the purpose of contributing to this repo. This [example](https://github.com/GoogleCloudPlatform/professional-services/blob/master/examples/cloud-composer-cicd/cloudbuild.yaml) from professional-services might be helpful

If you don't want to manage the container in GCR or want to be lazy for the purposes of this PR, then at least move this line into the cloud build.
For Example
```
- name: python:3.8.0-slim
  entrypoint: 'bash'
 args: ["python3 -m pip install -r udfs/tests/requirements.txt && udfs/tests/run.sh"]
```

Regardless we should add to contributing a step to install the dependencies in the virtualenv so for local testing.